src/__tests__/send-request.js

```diff
@@ -1,4 +1,72 @@
-import { encodePostData } from '../send-request'
+import { encodePostData, xhr } from '../send-request'
+
+const generateXmlHttpRequestMock = (status = 502) => ({
+    open: jest.fn(),
+    setRequestHeader: jest.fn(),
+    onreadystatechange: jest.fn(),
+    send: jest.fn(),
+    readyState: 4,
+    responseText: JSON.stringify('something here'),
+    status: status,
+})
+
+const generateXhrParams = (markRequestFailed, onXHRError = () => {}) => ({
+    url: 'https://any.posthog-instance.com',
+    data: '',
+    headers: {},
+    options: {},
+    captureMetrics: {
+        incr: () => {},
+        startRequest: () => {},
+        decr: () => {},
+        finishRequest: () => {},
+        markRequestFailed,
+    },
+    callback: () => {},
+    retriesPerformedSoFar: null,
+    retryQueue: {
+        enqueue: () => {},
+    },
+    onXHRError,
+})
+
+describe('when xhr requests fail', () => {
+    given('mockXHR', generateXmlHttpRequestMock)
+    given('markRequestFailed', jest.fn)
+
+    beforeEach(() => {
+        window.XMLHttpRequest = jest.fn(() => given.mockXHR)
+    })
+
+    it('does not error if the configured onXHRError is not a function', () => {
+        const onXHRError = {}
+        xhr(generateXhrParams(given.markRequestFailed, onXHRError))
```

Comment:
The way `given2` should work is that you should have `given('xhrParams', () => {...})`, `given('onXHRError', ...)` and all other mocks in the describe block, and then override the differences locally inside the tests. This instead of running some version of `generateXhrParams(given.markRequestFailed, onXHRError)` inside the tests.

In this case you'd do `given('onXHRError', () => ({}))` inside this test to override the function only here.